addons/point_of_sale/static/src/js/models.js

```diff
@@ -60,7 +60,11 @@ class PosModel {
      * don't want this default calculation of cid.
      * @param {Object?} defaultObj its props copied to this instance.
      */
-    constructor(defaultObj) {
+    constructor() {
+        this.setup(...arguments);
+    }
+    // To be used by Model patches to patch constructor
+    setup(defaultObj) {
```

Comment:
Correct, classes are in fact functions with a prototype property that contains the methods and the constructor is the function body, but you can't patch the function body. This is the same reason why Owl components have a setup method instead of using the constructor.